File: app/src/main/java/com/holike/cloudshelf/dialog/LoadingDialog.java
package com.holike.cloudshelf.dialog;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.Window;

import com.holike.cloudshelf.R;


public class LoadingDialog extends Dialog {

    public LoadingDialog(Context context) {
        super(context, R.style.LoadingDialogStyle);
        setCancelable(true);
        setCanceledOnTouchOutside(false);
        Window window = getWindow();
        if (window != null)
            window.setWindowAnimations(R.style.LoadingDialogWindowStyle);
    }

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_loading);
    }
}

File: app/src/main/java/com/holike/cloudshelf/dialog/LoadingDialog.kt
package com.holike.cloudshelf.dialog

import android.content.Context
import com.holike.cloudshelf.R
import pony.xcode.base.CommonDialog

class LoadingDialog(context: Context?) : CommonDialog(context, R.style.LoadingDialogStyle) {

    init {
        setCancelable(true)
        setCanceledOnTouchOutside(false)
    }

    override fun getLayoutResourceId(): Int = R.layout.dialog_loading
}